### PR TITLE
Rename QMatrixClient to Quotient

### PR DESCRIPTION
**Pull request by a-andreyev**
Since libQMatrixClient [was renamed](https://github.com/quotient-im/libQuotient/issues/325) to libQuotient

### CODE REVIEW

**Comment by Kaffeine** [comment]
I had some issues with Quotient compilation for Sailfish OS. I would consider it as a stopper that prevents us from update to Quotient.

**Comment by Kaffeine** [comment]
As of 014e6e42567b4eed2b6f551fc7dc92ed4fa068e4, the library version is not consistent across the commit:
- You use 0.5 in the RPM spec, but require 0.6 in CMakeLists
- `#if Q_MATRIX_CLIENT_VERSION >= Q_MATRIX_CLIENT_VERSION_CHECK(0, 2, 0)` makes no sense now

**Comment by a-andreyev** [comment]
Updated, so the app is not crashing (at the start at least :) and successfully flooding jolla-messages and notifications hub with the events.
No luck with kde-telepathy yet (probably due to my broken environment): getting notifications about new messages and logs with the data, but `ktp-contactlist` is segfaulting somewhere at the kpeople models.
Next step is probably to update the shared obs services for the deps repos. And also to filter room to direct chats only in the mer case similar to telepathy-morse logic.

**Comment by HenkKalkwater** [comment]
I'm not having problems with ktp-contactlist segfaulting, I've compiled this with the first beta release of libQuotient 6.0, if that makes any difference.

**Comment by a-andreyev** [comment]
Rebased read delivery status support from #19, introduced 'silent' hint for non-personal messages. Not tested yet.

File: src/messageschannel.cpp [hunk 1]
```diff
@@ -179,6 +197,13 @@ void MatrixMessagesChannel::processMessageEvent(const Quotient::RoomMessageEvent
     if (event->isRedacted())
         header[QStringLiteral("delivery-status")] = QDBusVariant(Tp::DeliveryStatusDeleted);
 
+    /* Read markers */
+    QList<Quotient::User*> usersAtEventId = m_room->usersAtEventId(event->id());
+    if ((usersAtEventId.contains(m_connection->matrix()->user()) && usersAtEventId.count()>1) ||
```
Comment (Kaffeine) [inline src/messageschannel.cpp:207]: I would suggest to replace this
```cpp
if ((usersAtEventId.contains(m_connection->matrix()->user()) && usersAtEventId.count()>1)       
  || (!usersAtEventId.contains(m_connection->matrix()->user()) && usersAtEventId.count()>0)) {
```
by something like this:
```cpp
const int selfReceipts = usersAtEventId.contains(m_connection->matrix()->user()) ? 1 : 0;
const hasOtherReceipts = usersAtEventId.count() > selfReceipts;
if (hasOtherReceipts) {
```

**Comment by a-andreyev** [comment]
Updated commits and installed with your patched `commhistory-daemon-0.8.18+kaffeine.wip` on Sailfish OS to support silent messages. It's beautiful! [:'))](https://i.imgflip.com/2olxvs.jpg)

**Pull request merged**